src/java/org/apache/cassandra/auth/AuthenticatedUser.java

```diff
@@ -74,6 +97,22 @@ public RoleResource getPrimaryRole()
         return role;
     }
 
+    /**
+     * The mode of authentication used to authenticate this user, e.g. 'password', 'mtls', 'anonymous', etc.
+     */
+    public String getMode()
+    {
+        return mode;
+    }
+
+    /**
+     * {@link IAuthenticator}-contextual metadata about how the user was authenticated.
```

Comment:
how it _was authentication_ or how _it iis going to be_ ?

Reply:
stupid question ... of course _how it was authenticated_ as this is in the class called "AuthenticatedUser" :D